### PR TITLE
Updates to dev branch

### PR DESCRIPTION
**Pull request by kacunningham413**
- Support presigned url image uploads
- Upload images in parallel
- Add basic auth via Secret Manager

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -29,4 +29,4 @@ COPY public ../public
 COPY src ../src
 ENV PORT=8080
```
Comment (ivanmkc) [inline Dockerfile:30]: I think this works as Cloud Run defaults to 8080, but you should probably use the environmental variable passed in by Cloud Run and not hardcode it here.

See https://cloud.google.com/run/docs/reference/container-contract#port

File: api/config.py [hunk 2]
```diff
@@ -1,4 +1,8 @@
-BASIC_AUTH_FORCE = False  # set to False to disable authorization
+import os
+
+BASIC_AUTH_FORCE = True  # set to False to disable authorization
+BASIC_AUTH_USERNAME = 'test-user'
+BASIC_AUTH_PASSWORD = os.getenv('BASIC_AUTH_PASSWORD', '')
```
Comment (ivanmkc) [inline api/config.py:5]: Is this used somewhere?

File: src/components/Form/Form.js [hunk 3]
```diff
@@ -10,23 +12,51 @@ const Form = (props) => {
...
       return;
     }
+    makeOpen(false);
     let formData = new FormData();
     const timestamp = Date.now();
     const date = new Date(timestamp);
-    for (const file in files) {
-      formData.append(file, files[file]);
+    var uploads = [];
+    for (let i = 0; i < files.length; i++) {
```
Comment (ivanmkc) [inline src/components/Form/Form.js:42]: You can simplify to:

```
for (const file in files) {
     uploads.push(uploadFile(file, jobId));
}
```

File: api/config.py (hunk 2 above)
Comment (kacunningham413) [reply api/config.py:5]: This is used by the Flask Basic Auth library via some magic...you are correct that it is not referenced anywhere else in the code, but it is actually configuring the authentication.

File: src/components/Form/Form.js (hunk 3 above)
Comment (kacunningham413) [reply src/components/Form/Form.js:42]: That loop construction what I initially tried, but it actually doesn't iterate over the files in the expected manner, so I needed to switch to the indexed version.

**Pull request merged**